src/Simplex/Messaging/Agent.hs

```diff
@@ -216,8 +223,9 @@ processCommand c@AgentClient {sndQ} st (corrId, connAlias, cmd) =
           removeSubscription c connAlias
           delConn
 
-    sendReplyQInfo :: SMPServer -> SndQueue -> m ()
-    sendReplyQInfo srv sq = do
+    sendReplyQInfo :: SndQueue -> m ()
+    sendReplyQInfo sq = do
+      srv <- getSMPServer
```

Comment:
I think it would be more pure to create server and rcvQueue in `joinConnection`, then "pass" `qInfo` to `sendReplyQInfo`.. what I mean is it'd be more semantically coherent with function names, and also symmetric to `createNewConnection` function